### ISSUE TITLE
quoted eval in a sub segfaults

### ISSUE THREAD

**Issue by toddr** [CpanelInc/perl-compiler#34]
```
$>perl -e'sub eval_in_a_sub { my $todo = shift; eval $todo; } eval_in_a_sub("print q{ok\n}")'         
ok
170622.155619 root@toddr.dev.cpanel.net (/usr/local/cpanel/B-C)
$>perlcc -S -g -e'sub eval_in_a_sub { my $todo = shift; eval $todo; } eval_in_a_sub("print q{ok\n}")'; ./a.out
zsh: segmentation fault (core dumped)  ./a.out

**Comment by toddr** [comment]
```
Program received signal SIGSEGV, Segmentation fault.
0x00007ffff7804c39 in S_pad_findlex (namepv=0x7fffffffd6f0 "&print", namelen=6, flags=0, cv=0x7ffff7bd3248 <PL_main_cv>, seq=508, Perl_warn=1, out_capture=0x7fffffffd148, out_name=0x7fffffffd2e0, out_flags=0x7fffffffd2d8) at pad.c:1140
1140	        const PADNAMELIST * const names = PadlistNAMES(padlist);


(gdb) bt
#0  0x00007ffff7804c39 in S_pad_findlex (namepv=0x7fffffffd6f0 "&print", namelen=6, flags=0, cv=0x7ffff7bd3248 <PL_main_cv>, seq=508, Perl_warn=1, out_capture=0x7fffffffd148, out_name=0x7fffffffd2e0, out_flags=0x7fffffffd2d8) at pad.c:1140
#1  0x00007ffff78054c3 in S_pad_findlex (namepv=0x7fffffffd6f0 "&print", namelen=6, flags=0, cv=0x6502d0 <sv_list+28752>, seq=510, Perl_warn=1, out_capture=0x7fffffffd148, out_name=0x7fffffffd2e0, out_flags=0x7fffffffd2d8) at pad.c:1286
#2  0x00007ffff78054c3 in S_pad_findlex (namepv=0x7fffffffd6f0 "&print", namelen=6, flags=0, cv=0x679e58, seq=1, Perl_warn=1, out_capture=0x0, out_name=0x7fffffffd2e0, out_flags=0x7fffffffd2d8) at pad.c:1286
#3  0x00007ffff780489b in Perl_pad_findmy_pvn (namepv=0x7fffffffd6f0 "&print", namelen=6, flags=0) at pad.c:971
#4  0x00007ffff77e6457 in Perl_yylex () at toke.c:6736
#5  0x00007ffff77fe38f in Perl_yyparse (gramtype=258) at perly.c:332
#6  0x00007ffff78e1e2b in S_doeval_compile (gimme=1 '\001', outside=0x6502d0 <sv_list+28752>, seq=510, hh=0x0) at pp_ctl.c:3383
#7  0x00007ffff78e4cc7 in Perl_pp_entereval () at pp_ctl.c:4181
#8  0x00007ffff788059b in Perl_runops_standard () at run.c:41
#9  0x00007ffff77ba477 in S_run_body (oldscope=1) at perl.c:2485
#10 0x00007ffff77b9f54 in perl_run (my_perl=0x6576c8 <BC_unthreaded_interpreter>) at perl.c:2408
#11 0x000000000040c1b7 in main (argc=1, argv=0x7fffffffe238, env=0x7fffffffe248) at /usr/local/cpanel/B-C/t/v5.24.1/C-COMPILED/bctestc/034-0.c:12547

**Comment by toddr** [comment]
Fixed by a570619ef3c5b1c68433c3e5dd82f181e64cda0a
